src/main/java/de/zalando/aruha/nakadi/exceptions/NoStreamingSlotsAvailable.java

```diff
@@ -0,0 +1,14 @@
+package de.zalando.aruha.nakadi.exceptions;
+
+import javax.ws.rs.core.Response;
+
+public class NoStreamingSlotsAvailable extends NakadiException {
+    public NoStreamingSlotsAvailable(final int totalSlots) {
+        super("No free slots for streaming available. Total slots: " + totalSlots);
+    }
+
+    @Override
+    protected Response.StatusType getStatus() {
+        return Response.Status.BAD_REQUEST;
```

Comment:
I see. I even took some time to look for other options but came to a similar conclusion as you. Maybe using status code is not enough to programatically handle this scenario in an elegant way. We probably should have additional fields in the problem object returned with some application code for the error. I say so because matching over a error message that is dynamically generated would be bad for client implementation.

Something like `error_code: no_slots_available`.

When I implemented the event validation, I missed the opportunity to add more structure to the error message and today I see that it imposes some limitations for clients to handle that programatically.

The error return values are not so well defined in the API as well.